src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -498,58 +500,79 @@ def create_connectedk8s(cmd, client, resource_group_name, cluster_name, correlat
     # Generate request payload
     cc = generate_request_payload(location, public_key, tags, kubernetes_distro, kubernetes_infra,
                                   enable_private_link, private_link_scope_resource_id, distribution_version,
-                                  azure_hybrid_benefit, enable_oidc_issuer, enable_workload_identity,
-                                  self_hosted_issuer)
+                                  azure_hybrid_benefit, enable_oidc_issuer, enable_workload_identity, gateway,
+                                  arc_agentry_configurations, arc_agent_profile, self_hosted_issuer)
 
     print("Step: {}: Azure resource provisioning has begun.".format(utils.get_utctimestring()))
     # Create connected cluster resource
     put_cc_response = create_cc_resource(client, resource_group_name, cluster_name, cc, no_wait)
+    dp_request_payload = put_cc_response.result()
     put_cc_response = LongRunningOperation(cmd.cli_ctx)(put_cc_response)
     print("Step: {}: Azure resource provisioning has finished.".format(utils.get_utctimestring()))
 
     # Checking if custom locations rp is registered and fetching oid if it is registered
     enable_custom_locations, custom_locations_oid = check_cl_registration_and_get_oid(cmd, cl_oid, subscription_id)
 
-    print("Step: {}: Starting to install Azure arc agents on the Kubernetes cluster.".format(utils.get_utctimestring()))
+    # Perform DP health check
+    _ = utils.health_check_dp(cmd, config_dp_endpoint)
+
+    # Retrieving Helm chart OCI Artifact location
+    helm_values_dp = utils.get_helm_values(cmd, config_dp_endpoint, release_train, request_body=dp_request_payload)
+
+    registry_path = os.getenv('HELMREGISTRY') if os.getenv('HELMREGISTRY') else \
+        helm_values_dp["repositoryPath"]
+
+    # Get azure-arc agent version for telemetry
+    azure_arc_agent_version = registry_path.split(':')[1]
+    telemetry.add_extension_event('connectedk8s',
+                                  {'Context.Default.AzureCLI.AgentVersion': azure_arc_agent_version})
+
+    # Get helm chart path
+    chart_path = utils.get_chart_path(registry_path, kube_config, kube_context, helm_client_location)
+
+    helm_content_values = helm_values_dp["helmValuesContent"]
+
+    # Substitute any protected helm values as the value for that will be null
+    for helm_parameter, helm_value in protected_helm_values.items():
+        helm_content_values[helm_parameter] = helm_value
+
+    print("Starting to install Azure arc agents on the Kubernetes cluster.")
```

Comment:
Can you please update to the new log type "print("Step: {}: ...")

Reply:
done